### PR TITLE
Migrate various tests to new assertions

### PR DESCRIPTION
**Pull request by martint**
## Release notes

(x) This is not user-visible and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestTryFunction.java [hunk 1]
```diff
@@ -54,35 +69,53 @@ public static long throwError()
...
-        assertFunction(createTryExpression("42"), INTEGER, 42);
-        assertFunction(createTryExpression("DOUBLE '4.5'"), DOUBLE, 4.5);
-        assertFunction(createTryExpression("DECIMAL '4.5'"), createDecimalType(2, 1), decimal("4.5", createDecimalType(2, 1)));
-        assertFunction(createTryExpression("TRUE"), BOOLEAN, true);
-        assertFunction(createTryExpression("'hello'"), createVarcharType(5), "hello");
-        assertFunction(createTryExpression("JSON '[true, false, 12, 12.7, \"12\", null]'"), JSON, "[true,false,12,12.7,\"12\",null]");
-        assertFunction(createTryExpression("ARRAY [1, 2]"), new ArrayType(INTEGER), asList(1, 2));
-        assertFunction(createTryExpression("TIMESTAMP '2020-05-10 12:34:56.123456789'"), createTimestampType(9), timestamp(9, "2020-05-10 12:34:56.123456789"));
-        assertFunction(createTryExpression("NULL"), UNKNOWN, null);
+        assertThat(assertions.expression("try(42)"))
```
Comment (dain) [inline core/trino-main/src/test/java/io/trino/operator/scalar/TestTryFunction.java:72]: Do you want these to be constants?

**Review by dain — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestTryFunction.java (hunk 1 above)
Comment (martint) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestTryFunction.java:72]: Yes, those get constant folded anyway, so not point in trying anything else. Also, `try` is special in that the inner expression cannot be pulled out of the call without changing its semantics.

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestFormatFunction.java [hunk 2]
```diff
@@ -13,66 +13,182 @@
...
 {
+    private QueryAssertions assertions;
+
+    @BeforeAll
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterAll
```
[the remark this answers is not in the document]
Comment (martint) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestFormatFunction.java:43]: No, there's no such a thing with junit. The afterAll methods get called unconditionally after all the tests complete, regardless of whether they succeed or fail.

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java [hunk 3]
```diff
@@ -47,1547 +50,3511 @@
...
+                .isEqualTo(123.0);
+
+        assertThat(assertions.function("ceiling", "-123.0E0"))
+                .isEqualTo(-123.0);
+
+        assertThat(assertions.function("ceiling", "123.45E0"))
+                .isEqualTo(124.0);
+
+        assertThat(assertions.function("ceiling", "-123.45E0"))
+                .isEqualTo(-123.0);
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java:337]: IntelliJ IDEA isn't happy about this and indicates a warning

```
'isEqualTo()' between objects of inconvertible types 'double' and 'ExpressionAssertProvider' 
```

this is obviously its fault, not the code's, but i still don't want people to start ignoring IDE warnings. They are helpful.

do you have any advice here?

**Pull request merged**